### PR TITLE
chore: add repo depot for mariner in AGC

### PR DESCRIPTION
**Pull request by UtheMan**
**What type of PR is this?**
Adds repo depot endpoints for mariner in air-gapped cloud environment for package updates.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/init-aks-custom-cloud-mariner.sh [hunk 1]
```diff
@@ -16,7 +16,12 @@ cp /root/AzureCACertificates/*.crt /etc/pki/ca-trust/source/anchors/
 
 cloud-init status --wait
 
-# TODO - Set the repoDepotEndpoint in a .repo file if package update becomes necessary
+marinerRepoDepotEndpoint="$(echo "${REPO_DEPOT_ENDPOINT}" | sed 's/\/ubuntu//')"
```
Comment (cameronmeissner) [inline parts/linux/cloud-init/artifacts/init-aks-custom-cloud-mariner.sh:19]: could we add an error for if this is empty in custom cloud case or something? might be a good sanity check

**Review by ganeshkumar5699 — APPROVED** [review]
Lgtm. If you can add the PR description too, that'd be nice for future reference :)

File: parts/linux/cloud-init/artifacts/init-aks-custom-cloud-mariner.sh (hunk 1 above)
Comment (UtheMan) [reply parts/linux/cloud-init/artifacts/init-aks-custom-cloud-mariner.sh:19]: @hbeberman from your side, what would be the expected behavior here if that happens? Wrap this around an if-block and don't update repoDepot endpoints? Or throw some sort of error message?

Comment (hbeberman) [reply parts/linux/cloud-init/artifacts/init-aks-custom-cloud-mariner.sh:19]: I think throwing an error would be preferrable, you don't want to leave the VMs in a state where they assume they'll be getting automatic package updates, but the repos aren't actually pointing to the correct mirrors.

File: parts/linux/cloud-init/artifacts/init-aks-custom-cloud-mariner.sh [hunk 2]
```diff
@@ -16,7 +16,16 @@ cp /root/AzureCACertificates/*.crt /etc/pki/ca-trust/source/anchors/
...
+if [[ "$marinerRepoDepotEndpoint" == "" ]]; then
```
Comment (UtheMan) [inline parts/linux/cloud-init/artifacts/init-aks-custom-cloud-mariner.sh:20]: wasn't sure if the fact that its empty is important enough to exit from the entire script. So far its just a warning that gets printed to logs.

**Review by hbeberman — APPROVED** [review]

**Pull request merged**